dask_ml/model_selection/_hyperband.py

```diff
@@ -0,0 +1,528 @@
+        self.cv_results_ = cv_results
+
+        self.multimetric_ = SHAs[best_bracket].multimetric_
+        raise gen.Return(self)
+
+    def metadata(self):
+        """Get information about how much computation is required for
+        :func:`~dask_ml.model_selection.HyperbandSearchCV.fit`. This can be called
+        before or after  :func:`~dask_ml.model_selection.HyperbandSearchCV.fit`.
+
+        Returns
+        -------
+        metadata : dict
+            Information about the computation performed by ``fit``. Has keys
+
+            * ``partial_fit_calls``, the total number of partial fit calls.
+            * ``estimators``, the total number of estimators created.
+            * ``brackets``, each of which has the same two keys as above.
+
+        Notes
+        ------
+        This algorithm runs several loops in an "embarassingly parallel"
+        manner. The ``brackets`` key represents each of these loops.
+
+        """
+        bracket_info = _hyperband_paper_alg(self.max_iter, eta=self.aggressiveness)
+        num_models = sum(b["estimators"] for b in bracket_info)
+        for bracket in bracket_info:
+            bracket["decisions"].update({1})
+            bracket["decisions"] = sorted(list(bracket["decisions"]))
+        num_partial_fit = sum(b["partial_fit_calls"] for b in bracket_info)
+        bracket_info = list(reversed(sorted(bracket_info, key=lambda x: x["bracket"])))
+
+        brackets = _get_hyperband_params(self.max_iter, eta=self.aggressiveness)
+        SHAs = self._get_SHAs(brackets)
+        for bracket in bracket_info:
+            b = bracket["bracket"]
+            bracket["SuccessiveHalvingSearchCV params"] = _get_SHA_params(SHAs[b])
+
+        info = {
+            "partial_fit_calls": num_partial_fit,
+            "estimators": num_models,
+            "brackets": {"bracket=" + str(b["bracket"]): b for b in bracket_info},
+        }
+        return info
+
+
+def _get_meta(hists, brackets, SHAs, key):
+    meta_ = {}
+    history_ = {}
+    for bracket in brackets:
+        hist = hists[bracket]
+
+        info_hist = {key(bracket, h["model_id"]): [] for h in hist}
+        for h in hist:
+            info_hist[key(bracket, h["model_id"])] += [h]
+        hist = info_hist
+        history_.update(hist)
+
+        calls = {k: max(hi["partial_fit_calls"] for hi in h) for k, h in hist.items()}
+        decisions = {hi["partial_fit_calls"] for h in hist.values() for hi in h}
+        meta_["bracket=" + str(bracket)] = {
+            "decisions": sorted(list(decisions)),
+            "estimators": len(hist),
+            "bracket": bracket,
+            "partial_fit_calls": sum(calls.values()),
+            "SuccessiveHalvingSearchCV params": _get_SHA_params(SHAs[bracket]),
+        }
+    return meta_, history_
+
+
+def _get_SHA_params(SHA):
+    return {
+        k: v
+        for k, v in SHA.get_params().items()
+        if "estimator_" not in k  # and k != "param_distribution"
```

Comment:
Why this if-statement?

Reply:
Now `_get_SHA_params` only returns the optional values of `SuccessiveHalvingSearchCV`. It does not include any parameters on the estimator or parameters.

This allows creation of a `SuccessiveHalvingSearchCV` object with the model and parameters also passed. The tests have been modified accordingly.